read_test.go

```diff
@@ -595,6 +595,78 @@ func TestReadPixelData_SkipPixelData(t *testing.T) {
 	}
 }
 
+// Returns a fake DICOM group 2 header with the FileMetaInformationGroupLength tag missing (0x0002,0x0000)
+func fakeHeader_NoFileMetaInformationGroupLength() *bytes.Buffer {
+	preamble := make([]byte, 128)
+	// First element after DICM is (0x0002,0x0002)
+	data := []byte{
+		0x44, 0x49, 0x43, 0x4d, 0x02, 0x00, 0x02, 0x00, 0x55, 0x49, 0x20, 0x00, 0x53, 0x65, 0x63, 0x6f,
+		0x6e, 0x64, 0x61, 0x72, 0x79, 0x20, 0x43, 0x61, 0x70, 0x74, 0x75, 0x72, 0x65, 0x20, 0x49, 0x6d,
+		0x61, 0x67, 0x65, 0x20, 0x53, 0x74, 0x6f, 0x72, 0x61, 0x67, 0x65, 0x00, 0x02, 0x00, 0x03, 0x00,
+		0x55, 0x49, 0x2c, 0x00, 0x31, 0x2e, 0x33, 0x2e, 0x36, 0x2e, 0x31, 0x2e, 0x34, 0x2e, 0x31, 0x2e,
+		0x33, 0x35, 0x31, 0x39, 0x30, 0x2e, 0x34, 0x2e, 0x31, 0x2e, 0x32, 0x30, 0x32, 0x31, 0x30, 0x36,
+		0x30, 0x38, 0x2e, 0x36, 0x30, 0x37, 0x37, 0x33, 0x33, 0x35, 0x34, 0x39, 0x35, 0x39, 0x33, 0x00,
+		0x02, 0x00, 0x10, 0x00, 0x55, 0x49, 0x14, 0x00, 0x31, 0x2e, 0x32, 0x2e, 0x38, 0x34, 0x30, 0x2e,
+		0x31, 0x30, 0x30, 0x30, 0x38, 0x2e, 0x31, 0x2e, 0x32, 0x2e, 0x35, 0x00, 0x02, 0x00, 0x12, 0x00,
+		0x55, 0x49, 0x20, 0x00, 0x31, 0x2e, 0x36, 0x2e, 0x36, 0x2e, 0x31, 0x2e, 0x34, 0x2e, 0x31, 0x2e,
+		0x39, 0x35, 0x39, 0x30, 0x2e, 0x31, 0x30, 0x30, 0x2e, 0x31, 0x2e, 0x30, 0x2e, 0x31, 0x30, 0x30,
+		0x2e, 0x34, 0x2e, 0x30, 0x08, 0x00, 0x18, 0x00, 0x55, 0x49, 0x2c, 0x00, 0x31, 0x2e, 0x33, 0x2e,
+		0x36, 0x2e, 0x31, 0x2e, 0x34, 0x2e, 0x31, 0x2e, 0x33, 0x35, 0x31, 0x39, 0x30, 0x2e, 0x34, 0x2e,
+	}
```

Comment:
How hard would it be to generate this "header" bytes in a more transparent fashion using the writeElement and similar utilities in write.go? I ask because it makes the test setup a little more transparent. 

e.g. build a slice of metadata Element structs and then pass to dicom.writeElement (internal API) to serialize them to a buffer?

If you think it might be too much of a pain, we can see about saving it as a follow up or something!

Reply:
Let me see how to do this but should be doable...

Reply:
Replaced by using your WriteElement methods to build the fake headers with and without the (0x2,0x0) tag.  Apologies for the delay.